taichi/transforms/type_check.cpp

```diff
@@ -244,6 +244,43 @@ class TypeCheck : public IRVisitor {
     return stmt;
   }
 
+  void insert_shift_op_assertion_before(Stmt *stmt, Stmt *lhs, Stmt *rhs) {
+    int rhs_limit = 64;
```

Comment:
A few comments.
1. `lhs` and `rhs` cannot be floats.
2. There's a `data_type_bits()` function.
3. In Taichi, runtime assertions are only inserted when `ti.init(debug=True)` for performance considerations.

Reply:
Great suggestions, Thanks!

Reply:
Done.